From fce0919880623a1a5a46c8ba84471181a743784b Mon Sep 17 00:00:00 2001
From: xplodwild <xplod@ouverta.fr>
Date: Mon, 23 Sep 2013 22:45:20 +0200
Subject: [PATCH] Setting for translucent statusbar on lockscreen

Change-Id: Ia28243b7980a836d1518216231fbf72581d834a0

Conflicts:
	res/values/custom_strings.xml
	res/xml/bars_and_menus_settings.xml
	src/org/omnirom/omnigears/BarsAndMenusSettings.java
---
 res/values/cm_strings.xml                           |  2 ++
 res/xml/status_bar.xml                              |  7 +++++++
 src/com/android/settings/cyanogenmod/StatusBar.java | 11 +++++++++++
 3 files changed, 20 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 4c1e2dc..906e83e 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -612,6 +612,8 @@
     <string name="status_bar_swipe_show_summary">Swipe down from the top of the screen to show status bar in fullscreen apps</string>
     <string name="status_bar_unhide_title">Status bar unhide</string>
     <string name="status_bar_unhide_summary">Status bar unhide</string>
+    <string name="status_bar_transparent_on_keyguard_title">Translucent lockscreen status bar</string>
+    <string name="status_bar_transparent_on_keyguard_summary">Makes the status bar translucent on the lockscreen</string>
 
     <!-- Setting checkbox summary for displaying USB debugging notification -->
     <string name="adb_notify">USB debugging notify</string>
diff --git a/res/xml/status_bar.xml b/res/xml/status_bar.xml
index a1c2139..65d611e 100644
--- a/res/xml/status_bar.xml
+++ b/res/xml/status_bar.xml
@@ -78,6 +78,13 @@
                 android:title="@string/status_bar_unhide_title"
                 android:summary="@string/status_bar_unhide_summary"
                 android:defaultValue="false" />
+
+        <CheckBoxPreference
+            android:key="status_bar_transparent_on_keyguard"
+            android:title="@string/status_bar_transparent_on_keyguard_title"
+            android:summary="@string/status_bar_transparent_on_keyguard_summary"
+            android:defaultValue="true" />
+
     </PreferenceCategory>
 
 </PreferenceScreen>
diff --git a/src/com/android/settings/cyanogenmod/StatusBar.java b/src/com/android/settings/cyanogenmod/StatusBar.java
index 6bdad40..545b3e8 100644
--- a/src/com/android/settings/cyanogenmod/StatusBar.java
+++ b/src/com/android/settings/cyanogenmod/StatusBar.java
@@ -40,10 +40,12 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
     private static final String STATUS_BAR_BATTERY = "status_bar_battery";
     private static final String STATUS_BAR_SIGNAL = "status_bar_signal";
     private static final String STATUS_BAR_CATEGORY_GENERAL = "status_bar_general";
+    private static final String STATUS_BAR_TRANSPARENT_ON_KEYGUARD = "status_bar_transparent_on_keyguard";
 
     private ListPreference mStatusBarAmPm;
     private ListPreference mStatusBarBattery;
     private ListPreference mStatusBarCmSignal;
+    private CheckBoxPreference mStatusBarTransparentOnKeyguard;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
@@ -103,6 +105,11 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
         if (Utils.isTablet(getActivity())) {
             generalCategory.removePreference(statusBarBrightnessControl);
         }
+
+        mStatusBarTransparentOnKeyguard = (CheckBoxPreference) prefSet.findPreference(STATUS_BAR_TRANSPARENT_ON_KEYGUARD);
+        mStatusBarTransparentOnKeyguard.setChecked(Settings.System.getInt(resolver,
+                Settings.System.STATUS_BAR_TRANSPARENT_ON_KEYGUARD, 1) == 1);
+        mStatusBarTransparentOnKeyguard.setOnPreferenceChangeListener(this);
     }
 
     @Override
@@ -126,6 +133,10 @@ public class StatusBar extends SettingsPreferenceFragment implements OnPreferenc
             Settings.System.putInt(resolver, Settings.System.STATUS_BAR_SIGNAL_TEXT, signalStyle);
             mStatusBarCmSignal.setSummary(mStatusBarCmSignal.getEntries()[index]);
             return true;
+        } else if (preference == mStatusBarTransparentOnKeyguard) {
+            boolean value = (Boolean) newValue;
+            Settings.System.putInt(resolver, Settings.System.STATUS_BAR_TRANSPARENT_ON_KEYGUARD, value ? 1 : 0);
+            return true;
         }
 
         return false;
-- 
1.8.4

